corehq/apps/accounting/utils/downgrade.py

```diff
@@ -70,6 +70,40 @@ def run_action(cls, only_downgrade_domain=None):
                     show_stack_trace=True
                 )
 
+    @classmethod
+    def _apply_downgrade_process(cls, oldest_unpaid_invoice, total, today, subscription):
```

Comment:
could `Downgrade` be turned into an instance where a lot of these arguments are passed in initially?